logbook-core/src/main/java/org/zalando/logbook/ExtendedLogFormatSink.java

```diff
@@ -0,0 +1,245 @@
+        return writer.isActive();
+    }
+
+    @Override
+    public void write(final Precorrelation precorrelation, final HttpRequest request) {
+        // nothing to do...
+    }
+
+    @Override
+    public void write(final Correlation correlation, final HttpRequest request,
+                      final HttpResponse response) throws IOException {
+
+        final ZonedDateTime startTime = correlation.getStart().atZone(timeZone);
+
+        // date
+        final String date = dateFormatter.format(startTime);
+
+        // time
+        final String time = timeFormatter.format(startTime);
+
+        // time-taken
+        final String timeTaken = BigDecimal
+                .valueOf(correlation.getDuration().toMillis())
+                .divide(BigDecimal.valueOf(1000), 3, RoundingMode.HALF_UP)
+                .toString();
+
+        // cs-protocol
+        final String csProtocol = request.getProtocolVersion();
+
+        // sc-bytes
+        final String scBytes = String.valueOf(response.getBody().length);
+
+        // cs-bytes
+        final String csBytes = String.valueOf(request.getBody().length);
+
+        // c-ip
+        final String clientIp = request.getRemote();
+
+        // s-dns
+        final String serverDns = request.getHost();
+
+        // sc-status
+        final String status = String.valueOf(response.getStatus());
+
+        // sc-comment
+        final String comment = response.getReasonPhrase();
+
+        // cs-method
+        final String method = request.getMethod();
+
+        // cs-uri
+        final String uri = request.getRequestUri();
+
+        // cs-uri-stem
+        final String uriStem = request.getPath();
+
+        // cs-uri-query
+        final String uriQuery;
+        if (!"".equals(request.getQuery())) {
+            uriQuery = "?" + request.getQuery();
+        } else {
+            uriQuery = OMITTED_FIELD;
+        }
+
+        Map<String, String> fieldValMap = new HashMap<>();
+        fieldValMap.put(Field.DATE.value, date);
+        fieldValMap.put(Field.TIME.value, time);
+        fieldValMap.put(Field.TIME_TAKEN.value, timeTaken);
+        fieldValMap.put(Field.CS_PROTOCOL.value, csProtocol);
+        fieldValMap.put(Field.SC_BYTES.value, scBytes);
+        fieldValMap.put(Field.CS_BYTES.value, csBytes);
+        fieldValMap.put(Field.CLIENT_IP.value, clientIp);
+        fieldValMap.put(Field.SERVER_DNS.value, serverDns);
+        fieldValMap.put(Field.RESP_STATUS.value, status);
+        fieldValMap.put(Field.RESP_COMMENT.value, comment);
+        fieldValMap.put(Field.REQ_METHOD.value, method);
+        fieldValMap.put(Field.REQ_URI.value, uri);
+        fieldValMap.put(Field.REQ_URI_STEM.value, uriStem);
+        fieldValMap.put(Field.REQ_URI_QUERY.value, uriQuery);
+
+        final List<String> outputFields = new ArrayList<>(fields);
+        final String output = outputFields.stream().map(outputField -> {
+                    if (supportedFields.contains(outputField)) {
```

Comment:
Would it be possible to create a private function and invoke it here, instead of this giant block? It is not as easy to read this way